### PR TITLE
feat(FEC-10768): expose in-stream DASH thumbnails

### PR DESCRIPTION
**Pull request by dan-ziv**
### Description of the Changes

* Add `DashThumbnailController` which controls and manage the image tracks.
* Add `getThumbnail(time: number): ?ThumbnailInfo` API implementation.
* Add parsing image tracks mechanism.
* Add `selectImageTrack` API implementation
* Add logic which optimise the manifest parsing on live manifests.

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/dash-adapter.js [hunk 1]
```diff
@@ -861,6 +907,23 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
...
+   * Get the parsed image tracks
+   * @function _getParsedImageTracks
+   * @returns {Array<ImageTrack>} - The parsed image tracks
+   * @private
+   */
+  _getParsedImageTracks(): Array<ImageTrack> {
+    if (this._manifestParser) {
+      const imageSet = this._manifestParser.getImageSet();
+      if (imageSet) {
+        this._thumbnailController = new DashThumbnailController(imageSet, this.src);
```
Comment (yairans) [inline src/dash-adapter.js:920]: in live it's already exist, can't you use the existing one?

File: src/dash-adapter.js [hunk 2]
```diff
@@ -1026,6 +1096,10 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
       }
       this._trigger(EventType.ERROR, new Error(error.severity, error.category, error.code, error.data));
       DashAdapter._logger.error(error);
+      // Stop all adapter processes on critical error
+      if (error.severity === Error.Severity.CRITICAL) {
+        this.destroy();
```
Comment (yairans) [inline src/dash-adapter.js:1101]: This should apply from the media source provider for all adapter

**Review by yairans — CHANGES_REQUESTED** [review]

File: src/dash-thumbnail-controller.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
+// @flow
+import {AdaptationSet} from './parser/adaptation-set';
```
Comment (Yuvalke) [inline src/dash-thumbnail-controller.js:2]: I think the prefix dash is redundant you can call it `ThumbnailController` cause you already on the dash adapter

**Review by yairans — DISMISSED** [review]

File: src/dash-thumbnail-controller.js (hunk 3 above)
Comment (dan-ziv) [reply src/dash-thumbnail-controller.js:2]: This is helpful for debugging and to find files easily (can be more than one thumbnail controller on a page).

**Pull request merged**